core/src/main/java/com/scalar/db/storage/dynamo/DynamoAdmin.java

```diff
@@ -1,78 +1,325 @@
+    this.metadataManager = metadataManager;
+    this.namespacePrefix = namespacePrefix;
+  }
+
   @Override
-  public void createNamespace(String namespace, Map<String, String> options)
-      throws ExecutionException {
-    throw new UnsupportedOperationException("implement later");
+  public void createNamespace(String namespace, Map<String, String> options) {
+    // In Dynamo DB storage, namespace will be added to table name as prefix along with dot
+    // separator.
   }
 
   @Override
   public void createTable(
       String namespace, String table, TableMetadata metadata, Map<String, String> options)
       throws ExecutionException {
-    throw new UnsupportedOperationException("implement later");
+    CreateTableRequest.Builder requestBuilder = CreateTableRequest.builder();
+
+    List<AttributeDefinition> columnsToAttributeDefinitions = new ArrayList<>();
+    makeAttribute(PARTITION_KEY, metadata, columnsToAttributeDefinitions);
+    if (!metadata.getClusteringKeyNames().isEmpty()) {
+      makeAttribute(CLUSTERING_KEY, metadata, columnsToAttributeDefinitions);
+      for (String clusteringKey : metadata.getClusteringKeyNames()) {
+        makeAttribute(clusteringKey, metadata, columnsToAttributeDefinitions);
+      }
+    }
+    if (!metadata.getSecondaryIndexNames().isEmpty()) {
+      for (String secondaryIndex : metadata.getSecondaryIndexNames()) {
+        makeAttribute(secondaryIndex, metadata, columnsToAttributeDefinitions);
+      }
+    }
+    requestBuilder.attributeDefinitions(columnsToAttributeDefinitions);
+
+    // build keys
+    buildPrimaryKey(requestBuilder, metadata);
+
+    // build local indexes that corresponding to clustering keys
+    buildLocalIndexes(namespace, table, requestBuilder, metadata);
+
+    // build secondary indexes
+    long ru = DEFAULT_RU;
+    if (options.containsKey(REQUEST_UNIT)) {
+      ru = Long.parseLong(options.get(REQUEST_UNIT));
+    }
+    buildGlobalIndexes(namespace, table, requestBuilder, metadata, ru);
+
+    // ru
+    requestBuilder.provisionedThroughput(
+        ProvisionedThroughput.builder().readCapacityUnits(ru).writeCapacityUnits(ru).build());
+
+    // table name
+    requestBuilder.tableName(getFullTableName(namespacePrefix, namespace, table));
+
+    // create table
+    try {
+      client.createTable(requestBuilder.build());
+      metadataManager.addTableMetadata(namespace, table, metadata);
+    } catch (Exception e) {
+      throw new ExecutionException("creating table failed", e);
+    }
+
+    try {
+      waitForTableCreation(client, getFullTableName(namespacePrefix, namespace, table));
+    } catch (DynamoDbException e) {
+      throw new ExecutionException("getting table description failed", e);
+    }
+
+    // scaling control
+    boolean noScaling = DEFAULT_NO_SCALING;
+    if (options.containsKey(NO_SCALING)) {
+      if (Boolean.parseBoolean(options.get(NO_SCALING))) {
+        noScaling = true;
+      }
+    }
```

Comment:
```suggestion
    boolean noScaling = Boolean.parseBoolean(options.getOrDefault(NO_SCALING, DEFAULT_NO_SCALING));
```

Reply:
Apply in [19938a0](https://github.com/scalar-labs/scalardb/pull/301/commits/19938a073c0e7ed103d9d226c6f128cbaa6cfdff)